### PR TITLE
TEST: fix race condition in Unit test

### PR DESCRIPTION
**Pull request by alfred2g**
Fix Race Condition In Unit Test

Description
-----------
A race condition could occur in UT where a thread frees a shared variable, then it is used by the main thread which could cause a core dump

Checklist:
----------

- [x ] I have tested my changes. No regression in existing tests.
- [ x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/unit_test/linux/CMakeLists.txt [hunk 1]
```diff
@@ -46,8 +46,8 @@
                 )
     set_target_properties(utils PROPERTIES
             LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib
-            COMPILE_FLAGS "-Wall -Og -pthread"
-            LINK_FLAGS "-Wall -Og -pthread -lpthread"
+            COMPILE_FLAGS "-Wall -Og -pthread -ggdb3"
```
Comment (gkwicker) [inline tests/unit_test/linux/CMakeLists.txt:49]: Why are we enabling debugging (`-ggdb3`) without using `CMAKE_CFLAGS_DEBUG`?  Ideally we should be executing unit tests both with and without debug flags and using [`CMAKE_BUILD_TYPE`](https://cmake.org/cmake/help/v3.0/variable/CMAKE_BUILD_TYPE.html) to control the build type.

Comment (alfred2g) [reply tests/unit_test/linux/CMakeLists.txt:49]: when would you not want debugging information with the UT?

Comment (gkwicker) [reply tests/unit_test/linux/CMakeLists.txt:49]: Ideally you will run your tests with and without debugging.  But the main issue here is that we're adding debugging support in a non-cmake standard way.  cmake gives you a standard method of doing this with `CMAKE_CFLAGS_DEBUG` and `CMAKE_BUILD_TYPE`, but we're not using that here.

Comment (alfred2g) [reply tests/unit_test/linux/CMakeLists.txt]: Ah I see what you mean, good point, pushed another commit, thanks!

Comment (gkwicker) [reply tests/unit_test/linux/CMakeLists.txt]: nice!

**Pull request merged**